src/libflake/flake/flake.cc

```diff
@@ -823,7 +823,11 @@ void initLib(const Settings & settings)
 {
     auto prim_getFlake = [&settings](EvalState & state, const PosIdx pos, Value * * args, Value & v)
     {
-        std::string flakeRefS(state.forceStringNoCtx(*args[0], pos, "while evaluating the argument passed to builtins.getFlake"));
+        NixStringContext context; // no context
+        auto flakeRefS = state.coerceToString(pos, *args[0], context,
+                                        "while evaluating the argument passed to builtins.getFlake",
+                                        false, false, true).toOwned();
```

Comment:
It seems that this now allows a derivation output to be passed, but it won't be realised, resulting in an error if the output isn't already in the store.

Reply:
Is that because of the `true` argument, or a general coerceToString property since it also converts `{ outPath = "foo"; }` to `"foo"`?

Reply:
It's a couple of things I guess. I think `coerceToPath` is a better match for this use case, although the `SourcePath` doesn't slot nicely into the flakeref parsing, but we shouldn't be using any flakeref parsing for a path value anyway.

It seems best to have a separate code path for when the argument's `.type()` is `nPath`.